src/snowflake/snowpark/file_operation.py

```diff
@@ -239,22 +239,31 @@ def put_stream(
         Returns:
             An object of :class:`PutResult` which represents the results of an uploaded file.
         """
-        if is_in_stored_procedure():
-            raise NotImplementedError(
-                "Stream uploads for stored procedure are not supported yet"
+        stage_location = stage_location.strip()
+        if not stage_location:
+            raise ValueError(
+                "stage_location cannot be empty. It must be a full stage path with prefix and file name like @mystage/stage/prefix/filename"
+            )
+        if stage_location[-1] == "/":
+            raise ValueError(
+                "stage_location should end with target filename like @mystage/prefix/stage/filename"
             )
-        else:
-
-            def parse_stage_file_location(stage_location: str):
-                stage_location = stage_location.strip()
-                if not stage_location:
-                    raise ValueError("stage_location cannot be empty")
-                elif stage_location[-1] == "/":
-                    raise ValueError("stage_location should end with target filename")
-                else:
-                    return stage_location.rsplit("/", maxsplit=1)
 
-            stage_with_prefix, dest_filename = parse_stage_file_location(stage_location)
+        if is_in_stored_procedure():
+            options = {
+                "parallel": parallel,
+                "source_compression": source_compression,
+                "auto_compress": auto_compress,
+                "overwrite": overwrite,
+            }
+            cursor = self._session._conn._cursor
```

Comment:
Do we also need the try-except similar to `put` so that the exception is wrapped?